core/models/templates.go

```diff
@@ -35,20 +36,32 @@ func (t *Template) FindTranslation(l i18n.Locale) *TemplateTranslation {
 }
 
 type TemplateTranslation struct {
-	Channel_        *assets.ChannelReference `json:"channel"`
-	Namespace_      string                   `json:"namespace"`
-	Locale_         i18n.Locale              `json:"locale"`
-	ExternalLocale_ string                   `json:"external_locale"`
-	Content_        string                   `json:"content"`
-	VariableCount_  int                      `json:"variable_count"`
+	Channel_        *assets.ChannelReference          `json:"channel"`
+	Namespace_      string                            `json:"namespace"`
+	Locale_         i18n.Locale                       `json:"locale"`
+	ExternalLocale_ string                            `json:"external_locale"`
+	Params_         map[string][]static.TemplateParam `json:"params"`
+	Content_        string                            `json:"content"`
 }
 
 func (t *TemplateTranslation) Channel() *assets.ChannelReference { return t.Channel_ }
 func (t *TemplateTranslation) Namespace() string                 { return t.Namespace_ }
 func (t *TemplateTranslation) Locale() i18n.Locale               { return t.Locale_ }
 func (t *TemplateTranslation) ExternalLocale() string            { return t.ExternalLocale_ }
 func (t *TemplateTranslation) Content() string                   { return t.Content_ }
-func (t *TemplateTranslation) VariableCount() int                { return t.VariableCount_ }
+func (t *TemplateTranslation) Params() map[string][]assets.TemplateParam {
+	prs := make(map[string][]assets.TemplateParam)
+
+	for k, v := range t.Params_ {
+		tprs := make([]assets.TemplateParam, len(v))
+		for i := range v {
+			tprs[i] = assets.TemplateParam(&v[i])
+		}
+		prs[k] = tprs
+	}
+
+	return prs
+}
```

Comment:
This is common challenge in go.. there's no free way to get from `[]s` to `[]S` even when `s implements S`.

What if we override `UnmarshalJSON` for `TemplateTranslation` so it creates a `map[string][]assets.TemplateParam` at load time? Rather than for every call to `Params()` ?

Reply:
Okay, I will change that